api/src/test/java/org/apache/iceberg/TestPartitionSpecValidation.java

```diff
@@ -327,12 +329,34 @@ public void testVariantUnsupported() {
         .hasMessage("Cannot partition by non-primitive source field: variant");
   }
 
+  @Test
+  public void testGeometryUnsupported() {
```

Comment:
Can these be collapsed with `testVariantUnsupported` in a `ParameterizedTest`?

Reply:
Can you also make sure that `identity` is tested like this?

Reply:
Refactored them as `TestIdentity.testUnsupported` and `TestPartitionSpecValidation.testUnsupported`.